src/providers/Web3ModalProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import type { ReactNode } from 'react';
+
+import '@rainbow-me/rainbowkit/styles.css';
+import { RainbowKitProvider, darkTheme } from '@rainbow-me/rainbowkit';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { WagmiProvider } from 'wagmi';
+
+import { config } from '../utils';
+
+type Props = {
+  children: ReactNode;
+};
+
+const queryClient = new QueryClient();
+
+export function Web3ModalProvider({ children }: Props) {
```

Comment:
are we using web3modal or it is legacy?

Reply:
legacy https://github.com/defi-wonderland/web3-react-boilerplate/pull/8/commits/a4a80bcb3d05cb213b045a2f40932f8ae160615c